src/main/java/net/snowflake/client/core/arrow/DateConverter.java

```diff
@@ -99,21 +118,42 @@ public String toString(int index) throws SFException {
     if (context.getDateFormatter() == null) {
       throw new SFException(ErrorCode.INTERNAL_ERROR, "missing date formatter");
     }
-    Date date = getDate(index, timeZoneUTC, false);
+    Date date =
+        getDate(
+            index,
+            TimeZone.getDefault(),
```

Comment:
previously we had there timeZoneUTC, not default timezone - should the new flag operate on this field also?